apps/test/unit/lib/kits/maker/PlaygroundComponentsTest.js

```diff
@@ -4,156 +4,555 @@ import Playground from 'playground-io';
+      // don't cover it with new tests.  If that happens, make sure you
+      // add matching tests below!
+      const components = createCircuitPlaygroundComponents(board);
       expect(Object.keys(components)).to.deep.equal([
-        '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'
+         'colorLeds',
+          'led',
+          'toggleSwitch',
+          'buzzer',
+          'soundSensor',
+          'tempSensor',
+          'lightSensor',
+          'accelerometer',
+          'buttonL',
+          'buttonR',
+          'touchPad0',
+          'touchPad1',
+          'touchPad2',
+          'touchPad3',
+          'touchPad6',
+          'touchPad9',
+          'touchPad10',
+          'touchPad12',
       ]);
-
-      // Every returned controller is a five.Led.RGB
-      Object.keys(components).forEach(key => {
-        expect(components[key]).to.be.an.instanceOf(five.Led.RGB);
-      });
-
-      // 20 sysex calls during initialization, two per LED.
-      for (let i = 0; i < 10; i++) {
-        // TODO (bbuchanan): Record what these calls mean.
-        expect(board.io.sysexCommand.getCall(i*2)).to.have.been.calledWith([0x40, 0x10, i, 0, 0, 0, 0]);
-        expect(board.io.sysexCommand.getCall(i*2+1)).to.have.been.calledWith([0x40, 0x11]);
-      }
     });
 
-    it('can initialize a second set of leds with a second board', () => {
+    it('can initialize a second set of components with a second board', () => {
       // Checks a necessary condition for a true johnny-five level reset.
       const boardOne = newBoard();
-      const controllers = initializeColorLeds(boardOne);
-      expect(controllers[0].board === boardOne).to.be.true;
-      expect(boardOne.io.sysexCommand.callCount).to.equal(20);
+      const componentsOne = createCircuitPlaygroundComponents(boardOne);
+      expect(componentsOne.led.board === boardOne).to.be.true;
 
       const boardTwo = newBoard();
-      const newControllers = initializeColorLeds(boardTwo);
-      expect(newControllers[0].board === boardTwo).to.be.true;
-      expect(boardTwo.io.sysexCommand.callCount).to.equal(20);
+      const componentsTwo = createCircuitPlaygroundComponents(boardTwo);
+      expect(componentsTwo.led.board === boardTwo).to.be.true;
     });
-  });
 
-  describe('initializeSoundSensor()', () => {
-    it('initializes one sensor', () => {
-      const sensor = initializeSoundSensor(board);
-      expect(sensor).to.be.an.instanceOf(five.Sensor);
-      // Doesn't use sysex at first
-      expect(board.io.sysexCommand.callCount).to.equal(0);
+    describe('colorLeds', () => {
+      it('creates an array of controllers', () => {
+        const {colorLeds} = createCircuitPlaygroundComponents(board);
+        expect(colorLeds).to.be.an.instanceOf(Array);
+        expect(colorLeds).to.have.length(10);
+      });
+
+      // Describe each Led by key/pin
+      [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].forEach(pin => {
+        describe(`colorLeds[${pin}]`, () => {
```

Comment:
Really?  I feel like in this case there's plenty to use to track it down.  You'll get the full test name with your failure message, something like:

```
Circuit Playground Components createCircuitPlaygroundComponents() colorLeds colorLeds[0] creates a five.Led.RGB
```
And I try to always group repeated dynamic-named tests like this in their own `describe` block so they don't get mixed into anything else.  Sure, you might see that whole name and decide to search for `colorLeds[0]` but when that didn't work you'd probably redirect to some other part of the name and be fine, or even `--grep "colorLeds[0]"` in a test run (which _will_ work) to get another look at exactly where in the hierarchy it exists.

On the other hand, the benefit of more usefully named tests with fewer assertions per test is that you often get a much more useful error message (this very specific test failed) than the assertion spew you'd get inside one test that covered the behavior of all ten LEDs.

I agree that it's a dangerous tool and we need to name things predictably, but that doesn't mean we should use it.